internal/server/rpc_handler.go

```diff
@@ -55,26 +55,31 @@ func (h *RPCHandler) ServeHTTP(writer http.ResponseWriter, req *http.Request) {
 
 	h.logger.Debug("Request received.", zap.String("method", req.Method), zap.String("path", req.URL.Path), zap.String("query", req.URL.RawQuery), zap.Any("body", requestBody))
 
-	upstreamID, respBody, resp, err := h.router.Route(ctx, requestBody)
-	if resp != nil {
-		defer resp.Body.Close()
-	}
+	upstreamID, jsonRPCRespBody, err := h.router.Route(ctx, requestBody)
 
 	if err != nil {
 		switch e := err.(type) {
-		case jsonrpc.DecodeError:
+		// Still pass the response to client if we're not able to decode response from upstream.
+		case *jsonrpc.DecodeError:
```

Comment:
i wonder how this changes the behavior on the indexers.  i'm assuming current decode errors get swallowed up into a generic 5XX or something, and the indexer will retry on the request.

once we add this change the indexer will potentially receive a poorly formatted JSON response.  hopefully it gracefully handles this scenario and retries.  then again this case is probably pretty rare.

Reply:
Yeah, I'm not totally sure of the consequences atm, but you're right, it's probably rare.